packages/components/pie-notification/src/defs.ts

```diff
@@ -1,12 +1,36 @@
 export const variants = ['neutral', 'neutral-alternative', 'info', 'success', 'warning', 'error'] as const;
 export const headingLevels = ['h2', 'h3', 'h4', 'h5', 'h6'] as const;
 
+export type ActionProps = {
+  /**
+   * The text to display inside the button.
+   */
+  text: string;
+
+  /**
+   * The ARIA label for the button.
+   */
+  ariaLabel?: string;
+
+  /**
+   * The optional action associated to the action.
+   *
+   * @returns void
+   */
+  onClick?: () => void;
```

Comment:
I am a bit unsure about this. Until now we have not accepted callback functions in our props because we want to move away from React patterns and stick to event based solutions. I.e. a component emits an event, the consumer chooses how to respond to it in their own code.

Can I ask why we have chosen to go with the callback approach here?

Reply:
I agree with @jamieomaguire here. We have a good example in the modal component of dispatching custom events which consumers can then listen for:

https://github.com/justeattakeaway/pie/blob/main/packages/components/pie-modal/src/index.ts#L225

Reply:
@xander-marjoram @jamieomaguire After today's call I finally understand how this event based approach works and I did some refactor in the component and now everything is working properly as well onClick has been removed out of this implementation. Thanks Jamie to take time to explain to me .